Runtime/Scripts/Internal/FFIClients/Pools/Memory/MemoryPool.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+
+namespace LiveKit.Internal.FFIClients.Pools.Memory
+{
+    public class MemoryPool : IMemoryPool
```

Comment:
This class should be thread-safe as we will be using the shared `instance` of `FFIBridge` to send requests unless we somehow can guarantee that we invoke it from a single background thread only (but it contradicts the singleton principle).
I propose you rely on `ArrayPool` - it already contains the synchronization on the chunk basis out of the box.

Reply:
I think it will be great to rid off static Singleton

WDYT?

Reply:
We can get rid of it but then we will need to have an instance of the FFIClient anyway and I believe it still will be in a single instance for multiple rooms and streams